Content.Server/CartridgeLoader/CartridgeLoaderSystem.cs

```diff
@@ -191,7 +191,13 @@ public bool InstallProgram(EntityUid loaderUid, string prototype, bool deinstall
         if (container.Count >= loader.DiskSpace)
             return false;
 
-        // TODO cancel duplicate program installations
+        if (TryComp(loader.CartridgeSlot.Item, out CartridgeComponent? loadedCartridge))
+            foreach (var program in GetInstalled(loaderUid))
+            {
+                if (TryComp(program, out CartridgeComponent? installedCartridge) && installedCartridge.ProgramName == loadedCartridge.ProgramName)
+                    return false;
+            }
+
```

Comment:
You should be doing this in `InstallCartridge`, not `InstallProgram`. Then you don't need to get the cartridge entity id from the loaders cartridge slot.
Also `InstallCartridge` is the method called when installing a cartridge.

Reply:
You can then use `Resolve` to get the `CartridgeComponent` instead of `TryComp` and just return when the cartridge entity doesn't have a cartridge component.

Reply:
Just assumed i should put it where the todo comment was, should be fixed now. 
Though I couldn't figure out how to use Resolve over TryComp without loosing the loadedCartridge variable.